src/main/java/net/sf/jabref/gui/importer/actions/CheckForNewEntryTypesAction.java

```diff
@@ -24,42 +31,105 @@
 public class CheckForNewEntryTypesAction implements PostOpenAction {
 
     @Override
-    public boolean isActionNecessary(ParserResult pr) {
-        Defaults defaults = new Defaults(BibDatabaseMode.fromPreference(Globals.prefs.getBoolean(JabRefPreferences.BIBLATEX_DEFAULT_MODE)));
-        BibDatabaseMode mode = new BibDatabaseContext(pr.getDatabase(), pr.getMetaData(), defaults).getMode();
-        // See if any custom entry types were imported, but disregard those we already know:
-        for (Iterator<String> i = pr.getEntryTypes().keySet().iterator(); i.hasNext();) {
-            String typeName = i.next().toLowerCase();
-            if (EntryTypes.getType(typeName, mode).isPresent()) {
-                i.remove();
-            }
-        }
-        return !pr.getEntryTypes().isEmpty();
+    public boolean isActionNecessary(ParserResult parserResult) {
+        return !getListOfUnknownAndUnequalCustomizations(parserResult).isEmpty();
     }
 
     @Override
-    public void performAction(BasePanel panel, ParserResult pr) {
-
-        StringBuilder sb = new StringBuilder();
-        sb.append(Localization.lang("Custom entry types found in file")).append(": ");
-        Object[] types = pr.getEntryTypes().keySet().toArray();
-        Arrays.sort(types);
-        for (Object type : types) {
-            sb.append(type).append(", ");
+    public void performAction(BasePanel panel, ParserResult parserResult) {
+
+        BibDatabaseMode mode = parserResult.getMetaData().getMode().orElse(Globals.prefs.getDefaultBibDatabaseMode());
```

Comment:
To avoid writing that method chain multiple times in this class, maybe extract a helper method?